// astro-skills/dist/routes/index-json.js
/**
 * GET /.well-known/skills/index.json
 *
 * Returns a JSON index of all available skills per the Agent Skills Discovery RFC.
 *
 * @see https://github.com/anthropics/anthropic-quickstarts/tree/main/skills-discovery-rfc
 */
export const GET = async () => {
    // Dynamic import of virtual module - resolved at runtime by Astro
    // @ts-expect-error - astro:content is a virtual module only available at runtime
    const { getCollection } = await import('astro:content');
    const skills = await getCollection('skills');
    const index = {
        skills: skills.map((skill) => {
            // Extract file paths from the files record and sort with SKILL.md first
            const filePaths = Object.keys(skill.data.files).sort((a, b) => {
                if (a === 'SKILL.md')
                    return -1;
                if (b === 'SKILL.md')
                    return 1;
                return a.localeCompare(b);
            });
            return {
                name: skill.data.name,
                description: skill.data.description,
                files: filePaths,
            };
        }),
    };
    return new Response(JSON.stringify(index, null, 2), {
        status: 200,
        headers: {
            'Content-Type': 'application/json',
            'Cache-Control': 'public, max-age=3600',
        },
    });
};
